account_vat_period_end_statement/models/account.py

```diff
@@ -454,26 +439,26 @@ def create_move(self):
             )
 
             move.line_ids = lines_to_create
-            move.post()
+            move.action_post()
             statement.state = "confirmed"
 
         return True
 
     def _add_end_debit_vat_data(self, lines_to_create, move, statement, statement_date):
         end_debit_vat_data = self._prepare_account_move_line(
             name="Tax Authority VAT",
-            account_id=statement.authority_vat_account_id,
-            move_id=move,
+            account_id=statement.authority_vat_account_id.id,
+            move_id=move.id,
             statement=statement,
             statement_date=statement_date,
-            partner_id=statement.authority_partner_id,
+            partner_id=statement.authority_partner_id.id,
         )
         if statement.authority_vat_amount > 0:
             end_debit_vat_data["credit"] = math.fabs(statement.authority_vat_amount)
             if statement.payment_term_id:
                 due_list = statement.payment_term_id.compute(
                     statement.authority_vat_amount, statement_date
-                )[0]
```

Comment:
come mai questa modifica? Il risultato di `compute ` è una lista (rif. https://github.com/odoo/odoo/blob/d75deff4791702e1f085b2b16247d7f7c5335720/addons/account/models/account_payment_term.py#L39)

Reply:
Perchè sulla 12 `compute` aveva `api.one` e chiamato così ritornava una lista di liste di tuple

Reply:
Ok grazie, non mi ero accorto che il risultato del metodo è in effetti una lista (`due_list`) su cui poi si cicla.